### DISCUSSION TITLE
When are kernels recomputed?

### DISCUSSION THREAD

**Discussion by celstark** [daducci/AMICO#157] [Q&A] [answered]
How much is done to check whether the bvals differ and how much variation is allowed?  The bval/bvecs I get off of a Bruker scanner always vary a bit.  It's always possible also that something did change at the scanner from one subject to the next that might have a qualitative change in the diffusion distribution.  I know the docs say that "scheme files with same b-values but different number/distribution of samples on each shell will result in the same precomputed kernels", which is great. But, what kind of tolerance is there on bvals and what kind of checking goes on to verify that it's within tolerance?

**Comment by nightwnvol** [comment, marked as answer]
Hi @celstark,
currently no check is made on bvals variation, simply if the kernels have already been computed, AMICO by default does not recompute them. You can force regeneration of kernels for different subjects each time by calling `generate_kernels(regenerate=True)`.
